internal/controller/environmentrequest_controller.go

```diff
@@ -222,20 +225,104 @@ func (r *EnvironmentRequestReconciler) reconcileEnvironmentProvider(ctx context.
 	return nil
 }
 
+func (r EnvironmentRequestReconciler) envVarListFrom(ctx context.Context, environmentrequest *etosv1alpha1.EnvironmentRequest) ([]corev1.EnvVar, error) {
+	etosEncryptionKey, err := environmentrequest.Spec.JobConfig.EtosEncryptionKey.Get(ctx, r.Client, environmentrequest.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	envList := []corev1.EnvVar{
+		{
+			Name:  "REQUEST",
+			Value: environmentrequest.Name,
+		},
+		{
+			Name:  "ETOS_API",
+			Value: environmentrequest.Spec.JobConfig.EtosApi,
+		},
+		{
+			Name:  "ETOS_GRAPHQL_SERVER",
+			Value: environmentrequest.Spec.JobConfig.EtosGraphQlServer,
+		},
+		{
+			Name:  "ETOS_ENCRYPTION_KEY",
+			Value: string(etosEncryptionKey),
+		},
+		{
+			Name:  "ETOS_ETCD_HOST",
+			Value: environmentrequest.Spec.JobConfig.EtosEtcdHost,
+		},
+		{
+			Name:  "ETOS_ETCD_PORT",
+			Value: environmentrequest.Spec.JobConfig.EtosEtcdPort,
+		},
+		{
+			// Optional when environmentrequest is not issued by testrun, i. e. created separately.
+			// When the environment request is issued by a testrun, this variable is propagated
+			// further from environment provider to test runner.
+			Name:  "ETR_VERSION",
+			Value: environmentrequest.Spec.JobConfig.EtosTestRunnerVersion,
+		},
+	}
+
+	var bus etosv1alpha1.RabbitMQ
+	for _, prefix := range []string{"", "ETOS_"} {
```

Comment:
This just obscures what is happening for me, better to just add all parameters as they are.
If function length is the problem then you can first create a map[string]string with all parameters:

```
cfg := map[string]string{
  "RABBITMQ_HOST": "asdfa",
  "ETOS_RABBITMQ_HOST": "asdfa2"
}
envList := []corev1.EnvVar{}
for name, value := range cfg {
  envList = append(envList, corev1.EnvVar{Name: name, Value: value})
}
```